sdk/resourcemanager/Azure.ResourceManager.Core/src/Operation/ArmOperation.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// Abstract class for long-running or synchronous applications.
+    /// </summary>
+    public abstract class ArmOperation : Operation
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ArmOperation"/> class for mocking.
+        /// </summary>
+        protected ArmOperation()
+        {
+        }
+
+        /// <summary>
+        /// Waits for the completion of the long running operations.
+        /// </summary>
+        /// <param name="cancellationToken"> A token to allow the caller to cancel the call to the service. The default value is <see cref="CancellationToken.None" />. </param>
+        /// <returns> A response with the <see cref="ArmOperation{TOperations}"/> operation for this resource. </returns>
+        /// <remarks>
+        /// <see href="https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-longrunning">Details on long running operation object.</see>
+        /// </remarks>
+        public virtual Response WaitForCompletion(CancellationToken cancellationToken = default)
+        {
+            return WaitForCompletion(OperationInternals<Response>.DefaultPollingInterval.Seconds, cancellationToken);
+        }
+
+        /// <summary>
+        /// Waits for the completion of the long running operations.
+        /// </summary>
+        /// <param name="pollingInterval"> The polling interval in seconds to check for status. </param>
+        /// <param name="cancellationToken"> A token to allow the caller to cancel the call to the service. The default value is <see cref="CancellationToken.None" />. </param>
+        /// <returns> A response with the <see cref="ArmOperation{TOperations}"/> operation for this resource. </returns>
+        /// <remarks>
+        /// <see href="https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-longrunning">Details on long running operation object.</see>
+        /// </remarks>
+        public virtual Response WaitForCompletion(int pollingInterval, CancellationToken cancellationToken = default)
+        {
+            while (true)
+            {
+                UpdateStatus(cancellationToken);
```

Comment:
The 'UpdateStatus' method throws if the token is set for cancellation?

Reply:
The definition for this was existing code.  I don't see i throwing when set for cancellation.  [The implementation is here](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/assets/Generator.Shared/OperationInternals.cs#L88)

Reply:
Normally you would preemptively call ThrowIfCancellationRequested at the start of the loop, but they call enough sdk methods that use the cancellation token in this method  for this to be moot.